nautobot/dcim/migrations/0030_migrate_region_and_site_data_to_locations.py

```diff
@@ -189,20 +247,60 @@ def migrate_site_and_region_data_to_locations(apps, schema_editor):
                 site_class=Site,
                 location_type_class=LocationType,
                 location_class=Location,
+                tagged_item_class=TaggedItem,
+                location_ct=location_ct,
+                site_ct=site_ct,
                 site_lt=site_lt,
                 exclude_lt="Region",
                 region_lt=region_lt,
             )
+            # make tag manager available in migration
+            for site_loc in SITE_TO_LOCATION_LOOKUP.values():
+                site_loc.tags = _NautobotTaggableManager(
+                    through=extras_models.TaggedItem, model=Location, instance=site_loc, prefetch_cache_name="tags"
+                )
+                site_loc.save()
+
+                # create an object change to document migration
+                ObjectChange.objects.create(
+                    action=extras_choices.ObjectChangeActionChoices.ACTION_UPDATE,
+                    change_context=extras_choices.ObjectChangeEventContextChoices.CONTEXT_ORM,
+                    change_context_detail="Migrated from Site",
+                    changed_object_id=site_loc.pk,
+                    changed_object_type=location_ct,
+                    object_data=serialize_object(site_loc),
+                    request_id=uuid.uuid4(),
+                )
     elif Site.objects.exists():  # Only Site instances exist, we make Site the top level LocationType
         site_lt = LocationType.objects.create(name="Site")
         add_location_contenttype_to_site_status_and_tags(apps, site_ct, location_ct)
         create_site_location_type_locations(
             site_class=Site,
             location_type_class=LocationType,
             location_class=Location,
+            tagged_item_class=TaggedItem,
+            location_ct=location_ct,
+            site_ct=site_ct,
             site_lt=site_lt,
             exclude_lt="Site",
         )
+        # make tag manager available in migration
+        for site_loc in SITE_TO_LOCATION_LOOKUP.values():
+            site_loc.tags = _NautobotTaggableManager(
+                through=extras_models.TaggedItem, model=Location, instance=site_loc, prefetch_cache_name="tags"
+            )
+            site_loc.save()
+
+            # create an object change to document migration
+            ObjectChange.objects.create(
+                action=extras_choices.ObjectChangeActionChoices.ACTION_UPDATE,
+                change_context=extras_choices.ObjectChangeEventContextChoices.CONTEXT_ORM,
+                change_context_detail="Migrated from Site",
+                changed_object_id=site_loc.pk,
+                changed_object_type=location_ct,
+                object_data=serialize_object(site_loc),
+                request_id=uuid.uuid4(),
+            )
```

Comment:
Can we create a helper function here? It looks like this is duplicated code between the "# Only Site instances exist` case and the "# Both Site and Region instances exist" case.